### PR TITLE
kanidm: 1.1.0-rc.15 -> 1.1.0-rc.16

### PR DESCRIPTION
**Pull request by Tom-Hubrecht**
## Description of changes

1.1.0-rc.16 Release Highlights

-    Replication for two node environments is now supported
-    Account policy supports password minimum length
-    Improve performance of webui
-    Add transitional compatability with SSSD
-    Improve TPM interfaces in unix clients
-    Allow importing more weak password schemes from FreeIPA
-    Support Attestation of Passkeys/Webauthn - this makes us the first IDM to support this!
-    Add entry-managed-by and hierarchial access control profiles
-    Rework and improve default access controls to further restrict default privileges
-    New replicated domain migration framework for distributed updates
-    Start to add PIV/Smartcard authentication groundwork
-    Allow changes to OAuth2 RS origin
-    Support RFC8414 OAuth2 metadata
-    Improve TLS error dialogs to assist administrators
-    Support RFC6749 Client Credentials Grant
-    Support custom claim maps in OIDC

C.f. https://github.com/kanidm/kanidm/releases/tag/v1.1.0-rc.16

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by erictapen — APPROVED** [review]
Thanks!

Ofborg has it's problems running the tests as usual, but I managed to build the NixOS test on the aarch64 community build box.

File: nixos/modules/services/security/kanidm.nix [hunk 1]
```diff
@@ -165,10 +165,17 @@ in
...
-        options.pam_allowed_login_groups = lib.mkOption {
-          description = lib.mdDoc "Kanidm groups that are allowed to login using PAM.";
-          example = "my_pam_group";
-          type = lib.types.listOf lib.types.str;
+        options = {
+          pam_allowed_login_groups = lib.mkOption {
+            description = lib.mdDoc "Kanidm groups that are allowed to login using PAM.";
+            example = "my_pam_group";
+            type = lib.types.listOf lib.types.str;
+          };
```
Comment (mweinelt) [inline nixos/modules/services/security/kanidm.nix:173]: import `lib.mkRenamedOptionModule` for this.

**Comment by mweinelt** [comment]
Package bumps should always carry the release notes URL in the commit message. Git commits and their messages will likely outlive GitHub.

**Comment by Tom-Hubrecht** [comment]
> Package bumps should always carry the release notes URL in the commit message. Git commits and their messages will likely outlive GitHub.
> 
> [`master`/pkgs/README.md#commit-conventions](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md?rgh-link-date=2024-02-07T13%3A09%3A55Z#commit-conventions)

Done, but as the release notes are stored on GitHub I'm not so sure about the utility :upside_down_face:

**Review by mweinelt — APPROVED** [review]

**Comment by adamcstephens** [comment]
> Done, but as the release notes are stored on GitHub I'm not so sure about the utility 🙃

It also helps us humans. It's much easier to click a link directly to the release notes.

**Pull request merged**